src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -66,6 +68,7 @@ protected function completeOrder(OrderInterface $order)
         $this->dispatchCheckoutEvent(SyliusOrderEvents::PRE_CREATE, $order);
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_PRE_COMPLETE, $order);
 
+        $this->applyTransition('complete', $order);
         $this->get('sm.factory')->get($order, OrderTransitions::GRAPH)->apply(OrderTransitions::SYLIUS_CREATE, true);
```

Comment:
Seems that this is some left-over here, or if not IMO should be refactored to new approach.